docs/modules/ROOT/pages/gsn.adoc

```diff
@@ -0,0 +1,93 @@
+
+== Receiving a relayed call
+
+We've mentioned how it is RelayHub, not your user, that actually ends up calling a function in your contract. We will refer to this as the _relayed call_. OpenZeppelin Contracts includes a number of utilities to make receiving relayed calls as easy as developing a regular Solidiy contract, without needing to worry about the low level details.
+
+The first step to writing a recipient is to inherit from our GSNRecipient contract. If you're also inheriting from other OpenZeppelin contracts, such as ERC20 or Crowdsale, this will work just fine: by adding GSNRecipient all of your token or crowdsale functions will also be GSN-callable.
+
+```solidity
+import "@openzeppelin/contracts/GSN/GSNRecipient.sol";
+
+contract MyContract is GSNRecipient, ... {
+
+}
+```
+
+=== msg.sender and msg.data
+
+There's only one extra care you should take when working with GSN recipient contracts: _you must never use `msg.sender` or `msg.data` directly_. On relayed calls, `msg.sender` will be `RelayHub` instead of your user! This doesn't mean however you won't be able to retrieve your users' addresses: `GSNRecipient` provides two functions, `_msgSender()` and `_msgData()`, which are drop-in replacements for `msg.sender` and `msg.data` while taking care of the low-level details. As long as you use these two functions instead of the original getters, you're good to go!
+
+WARNING: Third-party contracts you inherit from may not use these replacement functions, making them unsafe to use when mixed with `GSNRecipient`. If in doubt, head on over to our https://forum.openzeppelin.com/c/support[support forum].
+
+=== Accepting and charging
+
+Unlike regular contract function calls, each relayed call has an additional number of steps it must go through, which are functions of the `IRelayRecipient` interface `RelayHub` will call on your contract. `GSNRecipient` includes this interface but no implementation: most of writing a recipient involves handling these function calls. They are designed to provide flexibility, but basic recipients can safely ignore most of them while still being secure and sound.
+
+The OpenZeppelin Contracts provide a number of [tried-and-tested approaches]() for you to use out of the box, but you should still have a basic idea of what's going on under the hood.
+
+==== acceptRelayedCall
+
+First, RelayHub will ask your recipient contract if it wants to receive a relayed call. Recall that you will be charged for incurred gas costs by the relayer, so you should only accept calls that you're willing to pay for!
+
+There are multiple ways to make this work, including:
+ * having a whitelist of trusted users
```

Comment:
Bullet points need to be formatted.
"* having a whitelist of trusted users * only accepting calls to an onboarding function * charging users in tokens (possibly issued by you) * delegating the acceptance logic off-chain"

Reply:
Not sure what you mean by formatting, could you expand?

Reply:
The bullet points didn't format as bullet points in the preview of the document.  

The preview document displays as:
"* having a whitelist of trusted users * only accepting calls to an onboarding function * charging users in tokens (possibly issued by you) * delegating the acceptance logic off-chain"

Rather than:
* having a whitelist of trusted users 
* only accepting calls to an onboarding function 
* charging users in tokens (possibly issued by you) 
* delegating the acceptance logic off-chain

Reply:
Oh nice, thanks! I'll look into how lists are made in asciidoc. Where did you find the deploy preview? I though those were only generated for PRs to master.

Reply:
In Files Changed in the PR you can view the file and that shows the formatted document

https://github.com/OpenZeppelin/openzeppelin-contracts/blob/ecb569b23804ad3b56e700e59a4dff8e94a4a0a8/docs/modules/ROOT/pages/gsn.adoc

Reply:
Oh I see, I didn't know github rendered asciidoc. Thanks!